### PR TITLE
Add new metric magento_products_by_type_count_total

### PR DESCRIPTION
**Pull request by scolandrea**
This PR is adding a new metric `magento_products_by_type_count_total` to the extension. This metric gives a clear view of the count of products broken down by product type with the label product_type. 

Because of how helpful this metric is for projects I am working on now, I would be thankful if a new release of extension could be considered after adding this PR :)

### CODE REVIEW

**Comment by DavidLambauer** [comment]
Hi! Thank you for the PR. We'll look into this shortly!

File: src/Aggregator/Product/ProductByTypeCountAggregator.php [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    public function getType(): string
+    {
+        return 'gauge';
+    }
+
+    public function aggregate(): bool
+    {
+        $this->resetMetrics();
+
+        $connection = $this->resourceConnection->getConnection('products');
```
Comment (VladyslavSikailo) [inline src/Aggregator/Product/ProductByTypeCountAggregator.php:56]: There is no default connection with `products` name. Magento Catalog setup uses default connection, so I prefer to leave a default value here

**Review by VladyslavSikailo — CHANGES_REQUESTED** [review]
Hi @schmengler 

Thank you for your Pull Request!

Just a minor update is required, may I ask you to check it?

Thanks!

**Comment by schmengler** [comment]
> Hi @schmengler
> 
> Thank you for your Pull Request!
> 
> Just a minor update is required, may I ask you to check it?
> 
> Thanks!

I wondered if I submitted a PR in my sleep unknowingly :D that comment was for @scolandrea

File: src/Aggregator/Product/ProductByTypeCountAggregator.php (hunk 1 above)
Comment (scolandrea) [reply src/Aggregator/Product/ProductByTypeCountAggregator.php]: Done :)!

**Pull request merged**